bio-seq/src/codec/iupac.rs

```diff
@@ -149,24 +168,28 @@ impl SeqSlice<Iupac> {
     }
 }
 
-/*
 /// The complement of an IUPAC base is the reverse of the bit-pattern
-impl Complement for Iupac {
-    type Output = Self;
-
-    /// Clever bit-twiddling hack to reverse a pattern of 4 bits:
-    /// `(b * 0x11) >> 4 & 0x0f`
+impl ComplementMut for Iupac {
     fn comp(&mut self) {
-        *self = Iupac::unsafe_from_bits((*self as u8 * 0x11) >> 4 & 0x0f);
-    }
-
-    fn to_comp(&self) -> Self::Output {
-        let mut new = Iupac::unsafe_from_bits(*self as u8);
-        new.comp();
-        new
+        // Below are two methods:
+        // 1. Using a lookup table.
+        // 2. The 7 operation from https://graphics.stanford.edu/~seander/bithacks.html
+        // See: https://stackoverflow.com/questions/3587826/is-there-a-built-in-function-to-reverse-bit-order
+
+        // Use a lookup table
+        // *self = Iupac::unsafe_from_bits(IUPAC_COMPLEMENT_TABLE[*self as usize] >> 4 & 0x0f);
+
+        // Use the The 7 operation from
+        let b = *self as u32;
+        *self = Iupac::unsafe_from_bits(
```

Comment:
This is probably overkill for reversing just 4 bits, maybe it's worth it to start with the lookup table at least as a placeholder. We'll want `u32` and `u64` bit reversing for the Kmers, though, so we could use this there. Maybe to reverse the 1-bit degenerate encoding to start?